ACEHAL/dyn_utils.py

```diff
@@ -52,6 +52,58 @@ def __call__(self):
             # reject
             atoms.set_cell(orig_cell, True)
 
+class SwapMC:
+    """ASE trajectory attachment that does swap MC steps
+
+    Parameters
+    ----------
+    atoms: Atoms
+        atomic configuration
+    temperature_K: float
+        temperature in K
+    P_GPa: float
+        pressure in GPa
+    """
+    def __init__(self, atoms, temperature_K, P_GPa):
+        self.atoms = atoms
+
+        self.T = temperature_K
+        self.P = P_GPa
+
+        self.last_write_step = -1
+
+    def __call__(self):
+        """Do ASE dynamics trajectory attachment action"""
+
+        atoms = self.atoms
+
+        E_prev = atoms.get_potential_energy() + atoms.get_volume() * self.P * ase.units.GPa
+        
+        els = atoms.get_chemical_symbols()
+        ms = atoms.get_masses()
+
+        found = False
+        while found == False:
+            i1, i2 = np.random.randint(len(atoms), size=2) 
+            el1, el2 = els[i1], els[i2]
+            if el1 != el2:
+                found = True
+
+        m1, m2 = ms[i1], ms[i2]
```

Comment:
This won't switch arbitrary per-atom fields.  I suggest just switching the positions of the two atoms instead.